src/pal/src/misc/time.cpp

```diff
@@ -394,3 +394,40 @@ GetTickCount64()
     return retval;
 }
 
+/*++
+Function:
+  PAL_nanosleep
+
+Sleeps for the time specified in requestedTime.  If the operation is interrupted by a signal,
+remainingTime will contain the amount of time left from the original request.  Returns 0 on
+successful completion of the operation.
+--*/
+PALAPI
+INT
+PAL_nanosleep(
+    IN TIMESPEC *requestedTime,
+    OUT TIMESPEC *remainingTime
+    )
+{
+    struct timespec req;
+    struct timespec rem;
+    int result;
+
+    if(requestedTime == NULL)
+    {
+        return -1;
+    }
+
+    req.tv_sec = requestedTime->Seconds;
+    req.tv_nsec = requestedTime->Nanoseconds;
+
+    result = nanosleep(&req, &rem);
```

Comment:
I went ahead and fixed the implementation of PAL_nanosleep to handle interruptions.  I've also removed the remainingTime parameter.

I didn't do anything just yet with requestedTime.  I can convert the parameter to a long, but was trying to match the signature of nanosleep so that it is general use (vs later users having to make changes if they want to sleep for seconds vs. nanoseconds).  Ideally, no one sleeps for very long, of course.

Would you rather block callers from specifying multisecond sleeps, and if this is ever a requirement, the code will have to be fixed, or do you think we should leave it as is.  I don't have a strong opinion either way, but wanted to bring up my thoughts during implementation.